ttl/client/notification.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+
+	"github.com/pingcap/tidb/ddl/util"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+const ttlNotificationPrefix string = "/tidb/ttl/notification/"
+
+// NotificationClient is a client to notify other TTL workers
+type NotificationClient interface {
+	// Notify sends a notification
+	Notify(ctx context.Context, typ string, data string) error
+	// Watch opens a channel, in which we could receive all notifications
+	WatchNotification(ctx context.Context, typ string) clientv3.WatchChan
+}
+
+// Notify stores the corresponding K-V in the etcd
+func (c *EtcdClient) Notify(ctx context.Context, typ string, data string) error {
+	return util.PutKVToEtcd(ctx, c.etcdCli, 1, ttlNotificationPrefix+typ, data)
```

Comment:
It is strange that a common function is placed in the ```ddl/util```. I think it should be in the ```util```.

Reply:
I think it's because this function is only used by ddl (to notify after inserting ddl jobs). We could consider to extract this function to a more proper place (especially after implementing notification for the unified job/worker framework :thinking: ).

https://github.com/pingcap/tidb/issues/37119#issuecomment-1407316676

Reply:
I couldn’t agree more.